e2e/cypress/integration/integrations/outgoing_webhook/delete_outgoing_webhook_spec.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @integrations
+
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+
+let testTeam;
+
+describe('Integrations', () => {
+    before(() => {
+        // # Create test team, channel, and webhook
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+
+            // # Visit the webhook page
+            cy.visit(`/${team.name}/integrations/outgoing_webhooks/add`);
+        });
+    });
+
+    it('MM-T617 Delete outgoing webhook', () => {
+        // # Create an outgoing webhook with trigger word "testing"
+        cy.get('#displayName', {timeout: TIMEOUTS.ONE_MIN}).type('New Outgoing Webhook');
+        cy.get('#description').type('Description');
+        cy.get('#channelSelect').select('Town Square');
+        cy.get('#triggerWords').type('testing');
+        cy.get('#callbackUrls').type('https://outgoing-webhook-01snptsk6qvz.runkit.sh/');
+        cy.get('#saveWebhook').click();
+
+        //# Return to app and confirm outgoing webhook is working
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.uiPostMessageQuickly('testing', {timeout: TIMEOUTS.ONE_MIN});
+        cy.uiWaitUntilMessagePostedIncludes('Outgoing Webhook Payload');
+
+        // # Delete the webhook
+        cy.visit(`/${testTeam.name}/integrations/outgoing_webhooks`);
+        cy.findAllByText('Delete', {timeout: TIMEOUTS.ONE_MIN}).click();
+        cy.get('#confirmModalButton').click();
+
+        //# Return to app and confirm trigger word no longer works
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        cy.uiPostMessageQuickly('testing', {timeout: TIMEOUTS.ONE_MIN});
+
+        // Wait for BOT message
+        cy.wait(TIMEOUTS.TWO_SEC);
+        cy.getLastPostId().then((lastPostId) => {
```

Comment:
Not sure if there is a better way to assert that the bot message should not arrive?

Reply:
After you delete the webhook, you can verify via task if the hook url returns a 404 (or a 403 maybe) to make sure outgoing hook is deleted. You can do something similar below,

1.
Create a file `e2e/cypress/plugins/get_outgoing_webhook.js`,
```
const axios = require('axios');

module.exports = async ({url}) => {
    let response;

    try {
        response = await axios({method: 'get', url});
    } catch (err) {
        if (err.response) {
            response = err.response;
        }
    }

    return {status: response.status, data: response.data};
};
```

2. In `e2e/cypress/plugins/index.js`, add
```
...
const getOutgoingWebhook = require('./get_outgoing_webhook');
...
```

and

```
module.exports = (on, config) => {
    on('task', {
           ...
           getOutgoingWebhook,
           ...
```

3. In your test,
```
           cy.task('getOutgoingWebhook', {url: outgoingHookUrl}).then((res) => {
                expect(res.status).equal(404);
                expect(res.data.message).equal(message);
            });
```